src/inference/src/ie_core.cpp

```diff
@@ -885,22 +890,32 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         SetConfigForPlugins(any_copy(properties), device_name);
     }
 
-    Any get_property(const std::string& deviceName, const std::string& name, const AnyMap& arguments) const override {
-        OPENVINO_ASSERT(deviceName.find("HETERO:") != 0,
+    Any get_property_for_core(const std::string& name) const {
+        if (name == ov::force_tbb_terminate.name()) {
+            const auto flag = executorManager()->getTbbFlag();
+            return decltype(ov::force_tbb_terminate)::value_type(flag);
+        } else if (name == ov::cache_dir.name()) {
+            return ov::Any(coreConfig.get_cache_dir());
+        }
+        return ov::Any();
```

Comment:
Shouldn't it throw something instead of returning "just empty value"? How it is parsed later?

Reply:
agree

Reply:
Updated.